gradle-plugins/model-compiler/src/main/java/io/spine/gradle/compiler/util/UnknownOptions.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package io.spine.gradle.compiler.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.DescriptorProtos.DescriptorProto;
+import com.google.protobuf.DescriptorProtos.EnumDescriptorProto;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import com.google.protobuf.DescriptorProtos.ServiceDescriptorProto;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static java.util.Collections.emptyMap;
+
+/**
+ * A utility class which helps to get "unknown" Protobuf option field numbers
+ * (as in option type declaration) and option values.
+ *
+ * <p>For example, a map with pairs:
+ *
+ * <p>{@code 50123 -> "option_string_value_1"}
+ * <p>{@code 50124 -> "option_string_value_2"}
+ *
+ * <p>An option is "unknown" and serialized if there is no dependency on the artifact
+ * which contains the needed option definition.
+ * For example, we should not depend on "Spine/core-java" project artifacts to avoid
+ * circular dependency.
+ *
+ * <p>There can be several option properties, and several same numbers in the options string:
+ *
+ * [(decimal_max).value = "64.5", (decimal_max).inclusive = true];
+ *
+ * Currently, we do not need options with several properties.
+ * So, only first option property values are obtained, and others are ignored.
+ *
+ * @author Alexander Litus
+ */
+public class UnknownOptions {
+
+    private static final Pattern PATTERN_COLON = Pattern.compile(":");
+    @SuppressWarnings("HardcodedLineSeparator")
+    private static final Pattern PATTERN_NEW_LINE = Pattern.compile("\n");
```

Comment:
No the issues aren't related. It's just a simple way to split the multiline string into lines. 

Basically, `String.split(regexString)` does the same, but compiles the `regexString` into `Pattern` internally each time. Instead of doing that, we just compile the pattern ourselves, but once and for all.

Interestingly, whatever solution we use (including Guava's `Splitter`) we would end up with `"\n"` in the codebase anyway. So the suppression makes sense to me.